lib/suse/connect/api.rb

```diff
@@ -7,30 +7,34 @@ module Connect
     class Api
 
       class InvalidCredentialsError < StandardError; end
+      SYSTEM_UUID_LOCATION = 'config/system_uuid'.freeze
+      ROOT_PATH = Pathname.new(File.dirname(__FILE__)).join('../', '../', '../').freeze
+      CONNECT_API_URL = 'https://scc.suse.com/connect'.freeze
+      UUID_FILE_LOCATION = ROOT_PATH.join(SYSTEM_UUID_LOCATION).freeze
```

Comment:
Why create an extra file and all the boilerplate? We could just store it in the existing config file without any of this.
This would also allow the customer to set it to something more meaningful than a randomly generated string.

And why do we even need to generate a new identifier?
We could just use the fqdn, couldn't we? Since we have the SSL functionality, each RMT has one setup anyway, right? @ikapelyukhin

Reply:
@thutterer 
1. What's the relation between FQDN and SSL?
2. In regards to the value of RMT identifier, it depends on whether we want the user to change it or not. I was under the assumption that we want it to be as static as possible -- a separate file would survive DNS hostname changes and RPM package uninstallation/reinstallation.
    
    Making it changeable presents another issue -- once you change the ID, the old IDs will remain in SCC. Perhaps it would be better to assign a custom label to a uniquely-identified RMT in SCC, instead of changing the unique ID on RMT side?

Reply:
Don't you have to provide a hostname when generating a certificate for an RMT? So by design, each RMT has a hostname, that was set by its administrator. And if I put myself in that persons shoes, I would like to find that same RMT under that same name in my web interface SCC.

Because if you, let's say, setup 3 RMTs for your organization and sync them all with SCC at roughly the same time, you would see 3 new RMTs there, all with a unique Id, but you would still have no idea, which is which, because these Ids do not mean anything to you, and you have no good way to understand or change them.

Reply:
The FQDN can be obtained from the host system just by running `hostname`/`hostname --long`. FQDN doesn't have anything to do with SSL. Every host can have multiple hostnames.

Anyway, I don't have any strong preference of what the identifier should be.

Reply:
But if the RMT is registered itself (it should be in a supported scenario), the `SCC_` id will lead to its system + hwinfo which already holds the hostname.

Reply:
@digitaltom `SCCcredentials` isn't readable by `_rmt` user. Not without some sort of service startup hack, anyway.

Reply:
FYI: the identifier has to be an UUID or an UUID with `SCC_` prefix. see
https://github.com/SUSE/happy-customer/blob/f756108955e53f6fe546ddf1b0de78ebdc24ec8c/glue/app/controllers/api/connect/v3/organizations/base_controller.rb#L8
https://github.com/SUSE/happy-customer/blob/f756108955e53f6fe546ddf1b0de78ebdc24ec8c/glue/app/models/registration_proxy.rb#L29

Reply:
otherwise we will probably have to make changes (possibly harmful (?)) to the code which handles the header in 'glue'

Reply:
however what about adding a field 'fdqn' to `RegistrationProxy` in glue? and we could pass it in a separate header and show it to customer @ikapelyukhin @digitaltom @thutterer 
could probably be in a separate ticket thoough..